### PR TITLE
Remove PipelineParams

### PR DESCRIPTION
**Pull request by pivotal-nader-ziada**
The goal of this PR is to start a discussion about removing PipelineParams and moving its contents to PipelineRun.

In case we decide we want to do that, this is what things would look like

## Proposed changes
 * Move ServiceAccount and Results to PipelineRun

Fixes: #121

### CODE REVIEW

**Comment by pivotal-nader-ziada** [comment]
/cc @bobcatfish 
What do you think?

**Comment by pivotal-nader-ziada** [comment]
/hold

until we discuss and decide what we want to do

**Comment by bobcatfish** [comment]
Exciting! I'm defintiely all for moving the service account

I'm not sure about the results tho - i think it depends on how frequently we expect this to change. my preference would be to rename pipelineparams to something like `ResultsConfig` and leave the results stuff there - @tejal29 will probably have some opinions about how to handle this when she's back but @aaron-prindle might be able to weigh in as well

p.s. this should address #121 can i assign that one to you @pivotal-nader-ziada ? :D

**Comment by pivotal-nader-ziada** [comment]
I was thinking the results would be part of the runtime (pipelineRun) because you would change it to run in your own environment without having to change the original pipeline. You might also need to change it with every run if you want to keep history depending on how it will be implemented. 
Also I prefer to have less files to deal with as a user building/running a pipeline, having a file for the results location only seems like something that should be simplified 

@bobcatfish yes assign me the issue :)

File: pkg/apis/pipeline/v1alpha1/pipelinerun_types.go [hunk 1]
```diff
@@ -89,14 +92,37 @@ type PipelineRef struct {
...
+	ResultTargetTypeGCS = "gcs"
+)
+
+// Results tells a pipeline where to persist the results of runnign the pipeline.
+type Results struct {
+	// Runs is used to store the yaml/json of TaskRuns and PipelineRuns.
+	Runs ResultTarget `json:"runs"`
+
+	// Logs will store all logs output from running a task.
+	Logs ResultTarget `json:"logs"`
```
Comment (bobcatfish) [inline pkg/apis/pipeline/v1alpha1/pipelinerun_types.go]: while we're at it, why don't we simplify this and get rid of `Tests` and `Runs`? Discussions with @tejal29 indicated that we might treat Tests like a type of resource instead, and it's not clear how we'll use `Runs`

(sorry for the scope creep!!)

**Comment by bobcatfish** [comment]
> You might also need to change it with every run if you want to keep history depending on how it will be implemented.

I guess we have to see how #107 plays out!

I was assuming that the log config wouldn't change as often as some of the other Resources (e.g. git revisions) b/c you'd provide your bucket and we'd make sure we created separate dirs for each Run, etc. Maybe it'll be too hard to make that generic tho :thinking: 

Okay let's go ahead with this for now - we can always add back a new CRD for results if we need it for #017

**Comment by bobcatfish** [comment]
(feel free to remove the `hold` if you don't feel like removing `Tests` and `Runs` from results in this iteration)

**Comment by pivotal-nader-ziada** [comment]
/hold cancel

@bobcatfish made the changes to remove Runs and Tests, we can always add them again when we figure out more details

Please take another look
